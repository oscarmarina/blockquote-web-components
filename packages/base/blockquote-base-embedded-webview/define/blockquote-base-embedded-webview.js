import {BlockquoteBaseEmbeddedWebview} from '../src/BlockquoteBaseEmbeddedWebview.js';

window.customElements.define('blockquote-base-embedded-webview', BlockquoteBaseEmbeddedWebview);
