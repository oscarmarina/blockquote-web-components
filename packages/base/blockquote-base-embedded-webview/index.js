export { BlockquoteBaseEmbeddedWebview } from './src/BlockquoteBaseEmbeddedWebview.js';
export { BlockquoteBaseEmbeddedWebviewSize } from './src/BlockquoteBaseEmbeddedWebviewSize.js';
export { BlockquoteBaseEmbeddedWebviewResize } from './src/BlockquoteBaseEmbeddedWebviewResize.js';
export { BlockquoteBaseEmbeddedWebviewElement } from './src/BlockquoteBaseEmbeddedWebviewElement.js';
