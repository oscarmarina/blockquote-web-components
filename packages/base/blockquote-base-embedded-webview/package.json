{
  "name": "@blockquote-web-components/blockquote-base-embedded-webview",
  "version": "1.6.5",
  "description": "Webcomponent blockquote-base-embedded-webview following open-wc recommendations",
  "keywords": [
    "lit",
    "web-component",
    "lit-element"
  ],
  "license": "MIT",
  "author": "blockquote-base-embedded-webview",
  "type": "module",
  "exports": {
    ".": {
      "default": "./index.js"
    },
    "./package.json": {
      "default": "./package.json"
    },
    "./src/BlockquoteBaseEmbeddedWebview.js": {
      "default": "./src/BlockquoteBaseEmbeddedWebview.js"
    },
    "./src/BlockquoteBaseEmbeddedWebviewElement.js": {
      "default": "./src/BlockquoteBaseEmbeddedWebviewElement.js"
    },
    "./src/BlockquoteBaseEmbeddedWebviewResize.js": {
      "default": "./src/BlockquoteBaseEmbeddedWebviewResize.js"
    },
    "./src/BlockquoteBaseEmbeddedWebviewSize.js": {
      "default": "./src/BlockquoteBaseEmbeddedWebviewSize.js"
    },
    "./define/blockquote-base-embedded-webview-element.js": {
      "default": "./define/blockquote-base-embedded-webview-element.js"
    },
    "./define/blockquote-base-embedded-webview-resize.js": {
      "default": "./define/blockquote-base-embedded-webview-resize.js"
    },
    "./define/blockquote-base-embedded-webview-size.js": {
      "default": "./define/blockquote-base-embedded-webview-size.js"
    },
    "./define/blockquote-base-embedded-webview.js": {
      "default": "./define/blockquote-base-embedded-webview.js"
    },
    "./index.js": {
      "default": "./index.js"
    },
    "./src/styles/blockquote-base-embedded-webview-styles.css.js": {
      "default": "./src/styles/blockquote-base-embedded-webview-styles.css.js"
    },
    "./src/styles/blockquote-base-embedded-webview-element-styles.css.js": {
      "default": "./src/styles/blockquote-base-embedded-webview-element-styles.css.js"
    },
    "./src/styles/blockquote-base-embedded-webview-resize-styles.css.js": {
      "default": "./src/styles/blockquote-base-embedded-webview-resize-styles.css.js"
    },
    "./src/styles/blockquote-base-embedded-webview-size-styles.css.js": {
      "default": "./src/styles/blockquote-base-embedded-webview-size-styles.css.js"
    }
  },
  "main": "index.js",
  "module": "index.js",
  "files": [
    "/define/",
    "/src/",
    "index.js",
    "!/**/*.scss"
  ],
  "scripts": {
    "analyze": "cem analyze --litelement --globs \"{src,define}/**/*.{js,ts}\" \"index.js\"",
    "build": "echo \"This is not a TypeScript project, so no need to build.\"",
    "dev:vite": "vite build",
    "format": "npm run format:eslint && npm run format:prettier && npm run format:stylelint",
    "format:eslint": "eslint \"**/*.{js,ts,html}\" --fix --ignore-path .eslintignore",
    "format:prettier": "prettier \"**/*.{js,ts,json,html}\" --write --ignore-path .eslintignore",
    "format:stylelint": "stylelint \"**/*.{scss,css}\" --fix --allow-empty-input --ignore-path .eslintignore",
    "postinstall": "npm run sort:package",
    "lint": "npm run lint:eslint && npm run lint:prettier && npm run lint:stylelint",
    "lint:eslint": "eslint \"**/*.{js,ts,html}\" --ignore-path .eslintignore",
    "lint:prettier": "prettier \"**/*.{js,ts,json,html}\" --check --ignore-path .eslintignore",
    "lint:stylelint": "stylelint \"**/*.{scss,css}\" --allow-empty-input --ignore-path .eslintignore",
    "preview:vite": "vite preview",
    "sass:watch": "sass-style-template",
    "sort:package": "npx sort-package-json",
    "start": "concurrently -k -r \"npm:sass:watch\" \"npm:vite\"",
    "test": "wtr --coverage",
    "test:watch": "wtr --watch",
    "vite": "vite"
  },
  "lint-staged": {
    "**/*.{js,ts,html}": [
      "npm run format:eslint"
    ],
    "**/*.{js,ts,json,html}": [
      "npm run format:prettier"
    ],
    "**/*.{scss,css}": [
      "npm run format:stylelint"
    ]
  },
  "prettier": {
    "arrowParens": "avoid",
    "printWidth": 100,
    "singleQuote": true,
    "trailingComma": "all",
    "overrides": [
      {
        "files": "*.{scss,css}",
        "options": {
          "printWidth": 280,
          "singleQuote": false
        }
      }
    ]
  },
  "eslintConfig": {
    "parserOptions": {
      "ecmaVersion": "latest"
    },
    "extends": [
      "@open-wc",
      "prettier"
    ],
    "rules": {
      "class-methods-use-this": "off",
      "indent": [
        "error",
        2,
        {
          "SwitchCase": 1,
          "ignoredNodes": [
            "PropertyDefinition",
            "TemplateLiteral > *"
          ]
        }
      ],
      "no-unused-expressions": [
        "error",
        {
          "allowShortCircuit": true,
          "allowTernary": true
        }
      ],
      "object-curly-newline": "off",
      "import/extensions": [
        "error",
        "always",
        {
          "ignorePackages": true
        }
      ],
      "import/no-extraneous-dependencies": [
        "error",
        {
          "devDependencies": [
            "**/test/**/*.{js,ts}",
            "**/*.config.{js,ts,mjs,cjs}",
            "**/*.conf.{js,ts,mjs,cjs}"
          ]
        }
      ],
      "import/no-unresolved": "off",
      "import/prefer-default-export": "off",
      "lit/no-classfield-shadowing": "off",
      "lit/no-native-attributes": "off"
    }
  },
  "stylelint": {
    "extends": "stylelint-config-standard-scss",
    "rules": {
      "custom-property-pattern": null,
      "no-duplicate-selectors": null,
      "color-function-notation": null,
      "alpha-value-notation": null
    }
  },
  "dependencies": {
    "@blockquote/polymer": "^3.4.1",
    "lit": "^3.1.1"
  },
  "devDependencies": {
    "@blockquote-web-components/blockquote-base-common-dev-dependencies": "^1.7.2",
    "@blockquote-web-components/blockquote-foundations-sass": "^1.1.2",
    "@polymer/iron-test-helpers": "^3.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "customElements": "custom-elements.json"
}
