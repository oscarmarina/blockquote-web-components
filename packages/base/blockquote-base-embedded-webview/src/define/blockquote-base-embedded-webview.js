import {BlockquoteBaseEmbeddedWebview} from '../BlockquoteBaseEmbeddedWebview.js';

window.customElements.define('blockquote-base-embedded-webview', BlockquoteBaseEmbeddedWebview);
