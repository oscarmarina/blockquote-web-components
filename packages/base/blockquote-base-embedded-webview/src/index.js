export {BlockquoteBaseEmbeddedWebview} from './BlockquoteBaseEmbeddedWebview.js';
export {BlockquoteBaseEmbeddedWebviewSize} from './BlockquoteBaseEmbeddedWebviewSize.js';
export {BlockquoteBaseEmbeddedWebviewResize} from './BlockquoteBaseEmbeddedWebviewResize.js';
export {BlockquoteBaseEmbeddedWebviewElement} from './BlockquoteBaseEmbeddedWebviewElement.js';
