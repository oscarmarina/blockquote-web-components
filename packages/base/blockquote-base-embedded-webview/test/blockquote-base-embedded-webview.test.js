/* eslint-disable import/no-extraneous-dependencies */
import {html, fixture, assert, fixtureCleanup, oneEvent} from '@open-wc/testing';

import '../src/define/blockquote-base-embedded-webview.js';

suite('BlockquoteBaseEmbeddedWebview', () => {
  /**
   * @type {import('../src/index').BlockquoteBaseEmbeddedWebview}
   */
  let el;
  /**
   * @type {import('../src/index').BlockquoteBaseEmbeddedWebview}
   */
  let variant;

  teardown(() => fixtureCleanup());

  suite('Default', () => {
    setup(async () => {
      el = await fixture(html`
        <blockquote-base-embedded-webview heading="blockquote-base-embedded-webview">
          <template
            data-src="./test/test.html"
            data-option="Base"
            data-description="base - description"></template>
          <template
            data-src="./test/test2.html"
            data-option="Other"
            data-description="other - description"></template>
        </blockquote-base-embedded-webview>
      `);
      await el.updateComplete;
    });

    suite('Semantic Dom and a11y', () => {
      test('SHADOW DOM - Structure test', async () => {
        // @ts-ignore
        await assert.shadowDom.equalSnapshot(el, {ignoreAttributes: ['style', 'loading']});
      });

      test('LIGHT DOM - Structure test', async () => {
        // @ts-ignore
        await assert.lightDom.equalSnapshot(el, {ignoreAttributes: ['style', 'loading']});
      });

      test('a11y', async () => {
        await assert.isAccessible(el);
      });

      test('limit height', async () => {
        const elLimit = await fixture(html`
          <blockquote-base-embedded-webview limit-height heading="blockquote-base-embedded-webview">
            <template data-src="./test/test.html" data-option="Base Complex"></template>
          </blockquote-base-embedded-webview>
        `);
        // @ts-ignore
        await assert.shadowDom.equalSnapshot(elLimit, {ignoreAttributes: ['style', 'loading']});
      });

      test('aria level', async () => {
        const heading = el.shadowRoot?.querySelector('[aria-level]');
        el.headingLevel = 200;
        await el.updateComplete;
        assert.isTrue(heading?.getAttribute('aria-level') === '2');
      });

      test('select option load resource', async () => {
        const select = el.shadowRoot?.querySelector('select');
        const embedded = el.querySelector('blockquote-base-embedded-webview-element');
        if (select && embedded) {
          select.options[select.selectedIndex + 1].selected = true;
          await el.updateComplete;
          select.dispatchEvent(new CustomEvent('change'));
          const {detail} = await oneEvent(embedded, 'elementloaded', true);
          assert.isTrue(detail.title === select.options[select.selectedIndex].text);
        }
      });
    });
  });

  suite('variant', () => {
    setup(async () => {
      variant = await fixture(html`
        <blockquote-base-embedded-webview heading="blockquote-base-embedded-webview">
          <template data-src="./test/test.html"></template>
        </blockquote-base-embedded-webview>
      `);
      await variant.updateComplete;
    });

    suite('Semantic Dom and a11y', () => {
      test('SHADOW DOM - Structure test', async () => {
        // @ts-ignore
        await assert.shadowDom.equalSnapshot(variant, {ignoreAttributes: ['style', 'loading']});
      });

      test('LIGHT DOM - Structure test', async () => {
        // @ts-ignore
        await assert.lightDom.equalSnapshot(variant, {ignoreAttributes: ['style', 'loading']});
      });

      test('a11y', async () => {
        await assert.isAccessible(variant);
      });
    });
  });
});
