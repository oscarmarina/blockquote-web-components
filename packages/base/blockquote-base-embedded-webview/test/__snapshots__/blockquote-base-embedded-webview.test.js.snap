// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlockquoteBaseEmbeddedWebview > Behavior > limit height > SHADOW DOM 1`] = `
"<header>
  <div>
    <div
      aria-level="1"
      role="heading"
    >
      blockquote-base-embedded-webview
    </div>
    <div>
      <div class="select">
        <select aria-label="Cases">
          <option
            selected=""
            value="0"
          >
            Base Complex
          </option>
        </select>
      </div>
      <a
        class="open-externally"
        href="/test/test.html"
        target="_blank"
      >
        <span class="sr-only">
          View demo in a new tab
        </span>
        <span aria-hidden="true">
        </span>
      </a>
    </div>
    <p class="description">
    </p>
    <div
      aria-hidden="true"
      class="read-data-pos"
    >
      <span>
        0
      </span>
      <span>
        x
      </span>
      <span>
        0
      </span>
    </div>
  </div>
  <blockquote-base-embedded-webview-size>
  </blockquote-base-embedded-webview-size>
</header>
<div class="main">
  <blockquote-base-embedded-webview-resize>
    <slot name="embedded">
    </slot>
  </blockquote-base-embedded-webview-resize>
</div>
"
`;

exports[`BlockquoteBaseEmbeddedWebview > Default > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<blockquote-base-embedded-webview
  heading="blockquote-base-embedded-webview"
  heading-level="1"
>
  <template
    data-description="base - description"
    data-option="Base"
    data-src="/test/test.html"
  >
  </template>
  <template
    data-description="other - description"
    data-option="Other"
    data-src="/test/test2.html"
  >
  </template>
  <blockquote-base-embedded-webview-element slot="embedded">
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowfullscreen=""
      slot="embedded"
      src="/test/test.html"
      title="Base"
    >
    </iframe>
  </blockquote-base-embedded-webview-element>
</blockquote-base-embedded-webview>
"
`;

exports[`BlockquoteBaseEmbeddedWebview > Default > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<header>
  <div>
    <div
      aria-level="1"
      role="heading"
    >
      blockquote-base-embedded-webview
    </div>
    <div>
      <div class="select">
        <select aria-label="Cases">
          <option
            selected=""
            value="0"
          >
            Base
          </option>
          <option value="1">
            Other
          </option>
        </select>
      </div>
      <a
        class="open-externally"
        href="/test/test.html"
        target="_blank"
      >
        <span class="sr-only">
          View demo in a new tab
        </span>
        <span aria-hidden="true">
        </span>
      </a>
    </div>
    <p class="description">
      base - description
    </p>
    <div
      aria-hidden="true"
      class="read-data-pos"
    >
      <span>
        0
      </span>
      <span>
        x
      </span>
      <span>
        0
      </span>
    </div>
  </div>
  <blockquote-base-embedded-webview-size>
  </blockquote-base-embedded-webview-size>
</header>
<div class="main">
  <blockquote-base-embedded-webview-resize>
    <slot name="embedded">
    </slot>
  </blockquote-base-embedded-webview-resize>
</div>
"
`;

exports[`BlockquoteBaseEmbeddedWebview > variant > LIGHT DOM - Structure test > VARIANT - LIGHT DOM 1`] = `
"<blockquote-base-embedded-webview
  heading="blockquote-base-embedded-webview"
  heading-level="1"
>
  <template data-src="/test/test.html">
  </template>
  <blockquote-base-embedded-webview-element slot="embedded">
    <iframe
      allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
      allowfullscreen=""
      slot="embedded"
      src="/test/test.html"
      title="Demo"
    >
    </iframe>
  </blockquote-base-embedded-webview-element>
</blockquote-base-embedded-webview>
"
`;

exports[`BlockquoteBaseEmbeddedWebview > variant > SHADOW DOM - Structure test > VARIANT - SHADOW DOM 1`] = `
"<header>
  <div>
    <div
      aria-level="2"
      role="heading"
    >
      blockquote-base-embedded-webview
    </div>
    <div>
      <div class="select">
        <select aria-label="Cases">
          <option value="0">
            Base
          </option>
          <option
            selected=""
            value="1"
          >
            Other
          </option>
        </select>
      </div>
      <a
        class="open-externally"
        href="/test/test2.html"
        target="_blank"
      >
        <span class="sr-only">
          View demo in a new tab
        </span>
        <span aria-hidden="true">
        </span>
      </a>
    </div>
    <p class="description">
      other - description
    </p>
    <div
      aria-hidden="true"
      class="read-data-pos"
    >
      <span>
        0
      </span>
      <span>
        x
      </span>
      <span>
        0
      </span>
    </div>
  </div>
  <blockquote-base-embedded-webview-size>
  </blockquote-base-embedded-webview-size>
</header>
<div class="main">
  <blockquote-base-embedded-webview-resize>
    <slot name="embedded">
    </slot>
  </blockquote-base-embedded-webview-resize>
</div>
"
`;
